client/post-editor/editor-copy-post/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Internal dependencies
+ */
+import actions from 'lib/posts/actions';
+import { editPost } from 'state/posts/actions';
+import { setEditorPostId } from 'state/ui/editor/actions';
+import wpcom from 'lib/wp';
+
+function setPostCopy( post ) {
+	const postCopy = {
+		options: {
+			type: 'post',
+			content: post.content,
+		},
+		attributes: {
+			categories: Object.keys( post.categories ).map( key => post.categories[ key ].ID ),
```

Comment:
Lodash `_.map` could make this a bit cleaner:

```js
map( post.categories, 'ID' )
```

https://lodash.com/docs#map